app/logical/sources/alternates/deviantart.rb

```diff
@@ -0,0 +1,40 @@
+module Sources
+  module Alternates
+    class Deviantart < Base
+      BASE_HOSTS = %w[deviantart.com www.deviantart.com]
+
+      def force_https?
+        true
+      end
+
+      def domains
+        ["deviantart.com"]
+      end
+
+      def parse
+        # Add gallery link from old-style subdomain links
+        unless BASE_HOSTS.include?(@parsed_url.host) or @parsed_url.host.nil?
```

Comment:
Please don't use the written form for logic operators. Same thing for the other places you used them.

Also, personally I find `unless` harder to read than a simple if, so I try to invert the condition if easily possible. Like `if BASE_HOSTS.exclude?(@parsed_url.host) && @parsed_url.host.present?`. Most ruby/rails methods have an inverse of them to make this possible.